### PR TITLE
[SYCL][Graphs] Implement UR command-buffers

### PR DESCRIPTION
**Pull request by Bensuo**
- Move implementation of PI command buffers to UR

Note: there are currently some issues with a few e2e tests which I am investigating but may not be actually related to these changes.

Changes are developed with this PR in the Unified Runtime which defines command-buffers for UR: https://github.com/oneapi-src/unified-runtime/pull/560

PR is made against a temporary develop branch which has the UR L0 Adapter changes merged in to make it possible to review.

Once the changes have been reviewed we can update our develop branch and merge the changes in.

### CODE REVIEW

**Review by EwanC — APPROVED** [review]
Looks like an accurate port to me 😃

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+
+  if (auto Res = hQueue->executeCommandList(SignalCommandList, false, false))
+    return Res;
+
+  if (phEvent) {
+    *phEvent = RetEvent;
+  }
+
+  return UR_RESULT_SUCCESS;
+}
```
Comment (reble) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp]:
```suggestion
}

```

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  // Command list map so we can use queue::executeCommandList, TODO: Remove in
+  // future if possible
+  ur_command_list_map_t CommandListMap;
+  // Event which will signals the most recent execution of the command-buffer
+  // has finished
+  ur_event_handle_t SignalEvent = nullptr;
+  // Event which a command-buffer waits on until the wait-list dependencies
+  // passed to a command-buffer enqueue have been satisfied.
+  ur_event_handle_t WaitEvent = nullptr;
+};
```
Comment (reble) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.hpp]:
```suggestion
};

```

**Review by reble — APPROVED** [review]

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 3]
```diff
@@ -959,7 +959,7 @@ inline pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
     InfoType = UR_DEVICE_INFO_IMAGE_SRGB;
     break;
   case PI_EXT_ONEAPI_DEVICE_INFO_COMMAND_BUFFER_SUPPORT:
-    InfoType = (ur_device_info_t)UR_EXT_DEVICE_INFO_COMMAND_BUFFER_SUPPORT;
+    InfoType = (ur_device_info_t)UR_DEVICE_INFO_EXP_COMMAND_BUFFER_SUPPORT;
```
Comment (EwanC) [inline sycl/plugins/unified_runtime/pi2ur.hpp]: Noting that this needs removed to reflect https://github.com/oneapi-src/unified-runtime/pull/560/commits/412ebda36c5648609d749da03f520a7be3fed2e4 and a string returned from the device extension query as per https://github.com/oneapi-src/unified-runtime/pull/560#discussion_r1219760445

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp (hunk 1 above)
Comment (Bensuo) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.cpp]: Fixed thanks!

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.hpp (hunk 2 above)
Comment (Bensuo) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_command_buffer.hpp]: Fixed thanks!

File: sycl/plugins/unified_runtime/pi2ur.hpp (hunk 3 above)
Comment (Bensuo) [reply sycl/plugins/unified_runtime/pi2ur.hpp]: This has been removed now.

**Review by EwanC — APPROVED** [review]
Still LGTM after recent pushes

**Comment by Bensuo** [comment]
This is now rebased on top of an updated `sycl-graph-develop` so I'm going to merge now. We're having to use a temporary commit in my fork of UR because upstream DPC++ is not up-to-date with recent changes. Hopefully once the PR is up for these changes upstream will be more up-to-date.

**Pull request merged**